package-res/resources/web/vizapi/VizController.js

```diff
@@ -716,29 +716,31 @@ pentaho.VizController.getRgbGradientFromMultiColorHex = function(value, min, max
   var steps = colors.length-1;
   var range = max-min;
 
+  var startStep;
+  var endStep;
   if(range <= 0) {
-    var start = colors.length-1;
-    var end = start;
+    startStep = colors.length-1;
+    endStep = startStep;
   } else {
-    var start = Math.floor(((value-min)/range) * steps);
-    var end = Math.ceil(((value-min)/range) * steps);
+    var inRange = ((value-min)/range);
+    startStep = Math.floor(inRange * steps);
+    endStep = Math.ceil(inRange * steps);
   }
```

Comment:
Since both `startStep` and `endStep` are indexes of the `colors` array, we could rename them `startColorIndex` and `endColorIndex` so that future readers of this code endure fewer pains.